### PR TITLE
Release Variant Switcher v2

### LINKED ISSUE #4: Should have an option to disable recursion
#### Describe the current behavior

It recurses thru the entire frame tree

### PR DESCRIPTION
**Pull request by huayunh**
Fixes #4, #5, #6

#### Changes proposed in this Pull Request:
- See the changelog
- Need help on proofreading UI and changelog

#### Screenshots:
 
image

#### To Test:
- Follow the instruction in the readme (`yarn && yarn build`) after you link it up

### CODE REVIEW

File: variant-switcher/src/ui.tsx [hunk 1]
```diff
@@ -74,7 +99,27 @@ const App: React.FC = () => {
...
+                    onChange={(e) => {
+                        setDeepSwap(e.target.value === 'true' ? 'false' : 'true');
+                    }}
+                    name={'swapChild'}
+                    value={deepSwap}
+                    checked={deepSwap === 'true'}
                 />
+                <label htmlFor={'swapChild'}>
+                    Deep swap
+                    <div className={'hint-text'}>Swap child layers after parent instances are swapped</div>
```
Comment (huayunh) [inline variant-switcher/src/ui.tsx:121]: Looking for a better way to say this. 

Let's say that a component called Parent is built using instances of another component Child. Think of it as: 

```tsx
const Parent = (
    <div>
        <Child />
    </div>
)
```

And both Parent and Child are built with a prop called Theme, which can be either Dark or Light.

Now suppose dark parent is built using light child, and light parent is built using dark child. Think about the snack bar that is dark in light theme and light in dark theme.

Like this:
```tsx
<Parent Theme={'Light'}> === <div><Child Theme={'Dark'} /></div>
<Parent Theme={'Dark'}> === <div><Child Theme={'Light'} /></div>
```
Under these rules, if we try to switch from Theme=Light to Theme=Dark
- When "Deep Swap" is checked, a "light parent dark child" would become "dark parent, dark child".
- When "Deep Swap" is unchecked, a "light parent dark child" would become "dark parent, light child".

So back to the wording here. I think "Deep swap" accurately captures what it does, but the explanation text here is necessary. I'd like to say something like "attempt to swap child layers / instances even after its parents have been swapped to the specified 'To Variant'", but I couldn't find the best way to re-word it.

Comment (joebochill) [reply variant-switcher/src/ui.tsx:121]: Just thinking about this a little bit, I would want to have options something like:

```
PropertyName: string;
FromVariant: string;
ToVariant: string;

DeepSwitch: boolean; // (false = only switch top level items, true = also switch any nested children)
TwoWay: boolean; // (false = only switch 'from'->'to', true = also switch 'to'->'from')
```

I'm thinking of the `TwoWay` input as changing between a 'set' mode and a 'toggle' mode. You could even call it `mode` with two options: 'set' and 'toggle' if you want.

**Review by joebochill — COMMENTED** [review]
Code changes look fine. You might want to have Mariana try it out to give feedback on how it works.

**Comment by huayunh** [comment]
I can log an enhancement to `TwoWay`... there would be some work needed for this.

**Comment by joebochill** [comment]
> I can log an enhancement to `TwoWay`... there would be some work needed for this.

Isn't TwoWay what you are already doing:

> When "Deep Swap" is unchecked, a "light parent dark child" would become "dark parent, light child"

**Comment by huayunh** [comment]
No. Let me use a different example:

Suppose you have a `Parent` and a `Child` that both accept a prop called `color` that takes all kinds of common colors.

Suppose also that in your `Parent` preset you have:

- "red" `Parent` built with a "blue" `child`. 
- "green" `Parent` built with a "yellow" `child`. 

And you want to do a `from: undefined, to: green` on a "red" `Parent` "blue" `child` instance.

- `deepSwap = true`: result in green Parent green Child.
- `deepSwap = false`: result in green Parent yellow Child.

**Comment by huayunh** [comment]
I might be heading in the wrong direction and implemented a not-so-ideal function. Maybe we oughta have a video chat on this

**Comment by huayunh** [comment]
changing to 
```
Deep switch
Look into child layers after parent instances are switched
```

**Comment by joebochill** [comment]
Let's chat about this on Monday — i think it would be helpful to see some examples of what is happening rather than trying to describe them here

**Comment by huayunh** [comment]
Made some updates; here is what it looks like right now:
image

**Comment by huayunh** [comment]
Right now:

image

image

**Comment by huayunh** [comment]
Todo: add a fancy diagram to explain the recursion

File: variant-switcher/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## v2.0.0 (June 25, 2021)
```
Comment (joebochill) [inline variant-switcher/CHANGELOG.md:3]: date needs updated

File: variant-switcher/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+
+### Added
+
+-   When an instance is switched to the specified "To Variant", you can now choose to perform a "deep switch" to decide whether this plugin should keep looking into its child layers or not. This is helpful when you built dark-themed components using light-themed components, and don't want the plugin to "over-switch".
```
Comment (joebochill) [inline variant-switcher/CHANGELOG.md:7]:
```suggestion
-   When an instance is switched to the specified "To Variant", you can now choose to disable the "deep switch". Deep switch will look into all child layers and switch everything that matches the "From Variant". When this is disabled, the children of a switched element will not be switched. This is helpful when you built dark-themed components using light-themed components, and don't want the plugin to "over-switch".
```

**Review by joebochill — APPROVED** [review]

**Pull request merged**